classes/02class/exercises/c02-network07/gpavelar/README.md

```diff
@@ -0,0 +1,170 @@
+
+| route table              | destination | target              |
+| ------------------------ | ----------- | ------------------- |
+| devopsacademy-rt-public  | 0.0.0.0     | `devopsacademy-igw` |
+| devopsacademy-rt-private | 0.0.0.0     | `devopsacademy-ngw` |
+
+```bash
+# Create route - igw
+aws ec2 create-route --route-table-id rtb-09e3ebba72efa4af8 --destination-cidr-block 0.0.0.0/0 --gateway-id igw-04518145207146fe7
+# Output
+True
+
+# Create route - ngw
+aws ec2 create-route --route-table-id rtb-040e043e6f45c836c --destination-cidr-block 0.0.0.0/0 --gateway-id nat-0dcad8a11bc5686fa
+# Output
+An error occurred (InvalidGatewayID.NotFound) when calling the CreateRoute operation: The gateway ID 'nat-0dcad8a11bc5686fa' does not exist
+## I've already delted the natgatway created previously. But, Let's create it again
+
+### Creating NAT AGAIN
+## Allocate IP
+aws ec2 allocate-address --domain vpc
+
+# Output
+eipalloc-0711caa72a2042385      vpc     ap-southeast-2  52.62.142.50    amazon
+
+## SubnetId = subnet-08da4676baf658cb2
+## allocation-id = eipalloc-0711caa72a2042385
+## Create NAT gateway
+aws ec2 create-nat-gateway --allocation-id eipalloc-0711caa72a2042385  --subnet-id subnet-08da4676baf658cb2 --tag-specifications 'ResourceType="natgateway",Tags=[{Key="Name",Value="devopsacademy-ngw"}]'
+
+# Output
+7d2dc0a8-1b25-45ab-a711-b36b08e5f353
+NATGATEWAY      2020-07-06T15:54:11+00:00       nat-0c2e8f4ea792c8a75   pending subnet-08da4676baf658cb2        vpc-074c260483405f2bb
+NATGATEWAYADDRESSES     eipalloc-0711caa72a2042385
+TAGS    Name    devopsacademy-ngw
+
+aws ec2 delete-nat-gateway --nat-gateway-id nat-0c2e8f4ea792c8a75
+# Output
+nat-0c2e8f4ea792c8a75
+
+## Perform the command again
+aws ec2 create-route --route-table-id rtb-040e043e6f45c836c --destination-cidr-block 0.0.0.0/0 --gateway-id nat-0c2e8f4ea792c8a75
+# Output
+True
+```
+
+- Answer the following questions:
+
+  - How did you accomplish both instructions?
+
+    ```
+    I think so.
+    ```
+
+  - Why did you add 0.0.0.0 route to the IGW on public route table?
+
+    ```
+    Through this steps I enable my subnets to be public.
+    ```
+
+  - Why did you add 0.0.0.0 route to the NGW on private route table?
+
+    ```
+    Allow a way to connect my subnets securely over ther internet. Their IP addresses should definetely not go public. So we use NAT.
+    ```
+
+  - What is the difference of IGW and NGW?
+
+    ````
+    Basically, what I have got it, Internet Gateway is a way that my subnet has to communicate beyond the vpc (acessing internet). On the other hand, NAT just is a bridge betwwen your private subnet and the internet or any aws services.  ```
+
+    ````
+
+  - Can you delete the destination route to your VPC network? Why?
+
+    ```
+    You are not able to delete main route table.
+    If you create a custom subnet route table, You are able to delete just when there is no association in a custom route table.
```

Comment:
Your answer is for `Can you delete the main route table?` and thats correct.

The question here is -- when you create a route table it comes with one route by default for the local network. If you go to the console you will see that there is no way to "delete it". Why do you think they don't allow us to delete it?

Reply:
There are two things that I understood. 1st = default route table (that I considered as a main route table) 2nd = if you create a new route table (that i considered a custom one) you are able to delete it if there is no association to it.

So, I think it is not possible because you would lose connection to the instance. I could be wrong also. Maybe, I need to review this topic again.